app/src/main/java/net/gsantner/markor/util/AppSettings.java

```diff
@@ -572,6 +572,37 @@ public boolean isMultiWindowEnabled() {
         }
     }
 
+    public boolean isShowSearchResultOnCancel() {
+        return getBool(R.string.pref_key__is_show_search_result_on_cancel_enabled, true);
+    }
+
+    public Integer getSearchMaxDepth() {
+        Integer depth = getIntOfStringPref(R.string.pref_key__max_search_depth, Integer.MAX_VALUE);
+
+        if(depth == 0){
+            return Integer.MAX_VALUE;
+        }
+
+        return depth;
+    }
+
+    public List<String> getIgnoredSearchDirNames() {
+        List<String> ignoredDirs;
+        String pref = getString(R.string.pref_key__ignored_search_dir_names, "");
+        ignoredDirs =  Arrays.asList(pref.replace("\r", "").replace("\n\n", "\n").split("\n"));
+
+        return ignoredDirs;
+    }
+
+    public List<String> getIgnoredSearchFileNames() {
+        List<String> ignoredFiles;
+        String pref = getString(R.string.pref_key__ignored_search_file_names, "");
+        ignoredFiles = Arrays.asList(pref.replace("\r", "").replace("\n\n", "\n").split("\n"));
+
+        return ignoredFiles;
```

Comment:
It would be good to pre-populate these will patterns from RipGrep, SilverSearcher and FZF etc. At a minimum, we should not search inside `.git`

Reply:
.git and .thumbnails / thumbnails for sure